### DISCUSSION TITLE
Is there a way to pass options to whisper?

### DISCUSSION THREAD

**Discussion by PacoH** [absadiki/subsai#73] [Q&A]
I miss the default `--verbose True` behavior of whisper. Is there a way to pass this and other options to whisper from subsai CLI? Maybe using a JSON string like '{"verbose": "True"}' as in the API?

**Comment by absadiki** [comment]
@PacoH, Yes, it is already the case. You can use a JSON string in the CLI, like the following:
```shell
subsai ./assets/test1.mp4 --model openai/whisper --model-configs '{"model_type": "small", "verbose": "True"}'  --format srt
```

Let me know if you find any issues with it ?

**Reply by PacoH** [reply]
Oh, that is good to know. Thanks! How can I use other JSON strings? Is there a reference document that gives other examples? For example, I am trying different models and different model types. I'd like the output 'srt' file to have the filename plus model plus model type.

BTW I tried your method and it is working just fine with the `linto-ai/whisper-timestamped` model.

**Reply by absadiki** [reply]
Yes everything is documented on the [API docs](https://abdeladim-s.github.io/subsai/) .. In order to know what configs to use you just need to check the `config_schema` variable of the target model, for example [here](https://abdeladim-s.github.io/subsai/#subsai.models.whisper_model.WhisperModel.config_schema) are all the configs for the `openai/whisper` model.

For the output filename, I didn't implement the option to change the filename when using the CLI, because I thought it will be used just for batch processing .. But you can do it pretty easily using the web-UI and also using the package from Python.

You can use the following script for example:

```python
from subsai import SubsAI

file = './assets/test1.mp4'
model_name = "openai/whisper"
model_type = "base"
subs_ai = SubsAI()
model = subs_ai.create_model(model_name, {"model_type": model_type})
subs = subs_ai.transcribe(file, model)
subs.save(f'test1-{model_name}-{model_type}.srt')
```

**Reply by PacoH** [reply]
OK, that's the same as in the API reference except for the 'f' before the filename in `subs.save`, which I assume is a typo. But how do I run this from the command line on OS X? I would use the API but I really can't find any useful information on how to implement it on OS X. I am only familiar with REST API usage.

**Reply by absadiki** [reply]
The `f` is not a typo, it is Python syntax for string formatting :sweat_smile: 

To run it from the command line, you can save the script in a file and call it with the python interpreter, there are plenty of resources out there how to run a python script from the command line. 

But anyways, I added that option to the CLI for your convenience , you can now add a suffix to the output filename as follows:

```shell
subsai ./assets/test1.mp4 --model openai/whisper --model-configs '{"model_type": "small", "verbose": "True"}'  --format srt --output-suffix "openai-whisper-small" 
```

The output filename will be `test1-openai-whisper-small.srt`

Hope this helps!

**Reply by PacoH** [reply]
Wow, thank you for this addition.

First I upgraded openai-whisper and got: 

```
pip3 install -U openai-whisper

Installing collected packages: openai-whisper
  Attempting uninstall: openai-whisper
    Found existing installation: openai-whisper 20230124
    Uninstalling openai-whisper-20230124:
      Successfully uninstalled openai-whisper-20230124
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
subsai 1.2.0 requires openai-whisper==20230124, but you have openai-whisper 20230918 which is incompatible.
Successfully installed openai-whisper-20230918
```

Then I tried to upgrade subsai and got:

```
pip3 install -U git+https://github.com/abdeladim-s/subsai

Installing collected packages: openai-whisper
  Attempting uninstall: openai-whisper
    Found existing installation: openai-whisper 20230918
    Uninstalling openai-whisper-20230918:
      Successfully uninstalled openai-whisper-20230918
Successfully installed openai-whisper-20230124

subsai --version
subsai 1.2.0
```

20230124 is a much older version than 20230918. Can you possibly update the tool to use the latest version?

**Reply by absadiki** [reply]
I can update the package to the latest `openai-whisper` version, but the problem is it will lead to so many conflicts because other packages (like `whisper-timestamps` and `whisperX`) depend on different versions of whisper, so if you update just `openai-whisper`, other packages won't work. 
Besides, there are no major differences between the two versions I think, so I thought it is better to just stick with the actual version.
